### DISCUSSION TITLE
sunbird-rc-core dependency on keycloak

### DISCUSSION THREAD

**Discussion by prasadmoka** [Sunbird-RC/community#114] [Issues]
@dileepbapat @parthlawate  I wanted to use sunbird-rc-core alone without any dependency with keycloak. As per my observation as of now sunbird-rc-core is dependent on keycloak. Just wanted to check whether if there is any repo of sunbird-rc-core without any keycloak features so that we can use only core api's with our own schema.
Please let us know if you have any information on removing dependency with keycloak.
cc: @maheshkumargangula @vrayulu

**Comment by gamemaker1** [comment]
Hi @prasadmoka,

Keycloak is an auth provider for the Sunbird RC implementation and is required for the following:

- creating, authenticating and authorising entities
- attestation flow
- consent flow

Since these are all core features of the registry, not sure if you can completely remove the need for an auth provider like Keycloak.

---

If you are asking whether keycloak can be replaced by another auth provider, that should be technically possible, but I haven't really tried this out yet.

**Comment by pramodkvarma** [comment]
I am curious @prasadmoka. Can you please explain what you are trying to do? Would love to know the use case you are trying to address. Thanks.

**Reply by prasadmoka** [reply]
I am from Sanketika Consulting, we are working on setting up a registry for an agri-tech company. We are planning to use the sunbird-rc-core for registering the agriculture devices and farms. As part of initial release, all these registry APIs will be invoked by internal systems and we wanted to see how we can bypass the dependency with keycloak. In the subsequent releases we are planning to integrate with their in-house authentication and authorization process for all of these API calls.

**Reply by dileepbapat** [reply]
Created https://github.com/Sunbird-RC/community/issues/131  for it, will have it implemented soon

**Reply by dileepbapat** [reply]
https://github.com/Sunbird-RC/community/issues/131 fixed to enable registry with authentication disabled.
